CoreUtilities/src/au/gov/asd/tac/constellation/utilities/svg/SVGData.java

```diff
@@ -0,0 +1,355 @@
+     * @param attributeKey
+     * @return attributeValue
+     */
+    public final String getAttributeValue(final String attributeKey) {
+        return this.attributes.get(attributeKey);
+    }
+    
+    /**
+     * Sets the content of this SVGData.
+     * Content is considered plain text that lies within a SVG element 
+     * such as is the case with text elements.
+     * @param content
+     */
+    public void setContent(final String content) {
+        this.content = content;
+    }
+    
+    /**
+     * Returns the value of the content of this SVGData.
+     * @return content
+     */
+    public final String getContent() {
+        return this.content;
+    }
+    
+    /**
+     * Generates a string representation of SVG data captured within this object.
+     * will be formatted with indentations and line breaks to be written 
+     * directly to an output file.
+     * @return String in an SVG format.
+     */
+    @Override
+    public final String toString() {
+        cleanAttributes();
+        return toString(null);
+    }
+    
+    /**
+     * Prepares SVGData for export. 
+     * Attributes containing null values are removed.
+     */
+    public void cleanAttributes() {
+        this.attributes.values().removeIf(Objects::isNull);
+        this.getAllChildren().forEach(SVGData::cleanAttributes);
+    }
+    
+    /**
+     * Recursive function to generate a string equivalent of complex SVG data 
+     * captured within this object.
+     * @param prefix
+     * @return String representation of the current element and all of it's child elements.
+     */
+    private String toString(final String prefix) {
+        final StringBuilder svgString = new StringBuilder();
+        if (this.children.isEmpty() && this.content == null) {
+            svgString.append(elementToSVG(prefix));
+        } else {
+            svgString.append(elementHeaderToSVG(prefix));
+            if (this.children.isEmpty()) {
+                svgString.append(content);
+            } else {
+                svgString.append(elementChildrenToSVG(prefix));
+            }
+            svgString.append(elementFooterToSVG(prefix));
+        }
+        return svgString.toString();
+    }
+    /**
+     * Generates an "inline" SVG element.
+     * @param prefix represents indentation prefixes for the element.
+     * @return String representation of the current element
+     */
+    private String elementToSVG(final String prefix) {
+        final StringBuilder attributeBuilder = new StringBuilder();
+        final String linePrefix = SeparatorConstants.NEWLINE + (prefix == null ? "" : prefix);
+        final Set<String> keys = attributes.keySet();
+        keys.forEach(key -> {
+            if (SVGAttributeConstants.NAME_SPACE.getName().equals(key) && this.parent != null) {
+                //do nothing here
+            } else {
+                attributeBuilder.append(String.format(" %s=\"%s\"", key, attributes.get(key)));
```

Comment:
would it be easier to read if you were to flip this which would remove the need for an else clause?